### PR TITLE
Move upgrade-rollback job to use step registry

### PR DESCRIPTION
**Pull request by jluhrsen**
Signed-off-by: Jamo Luhrsen <jluhrsen@gmail.com>

### CODE REVIEW

**Comment by alvaroaleman** [comment]
/uncc
As this doesn't seem to be ready yet

**Comment by jluhrsen** [comment]
@alvaroaleman , I think it's ready now, but it's failing pj-rehearse which I think is expected at this point. You noted in another PR I have that UNRESOLVED_CONFIG doesn't work for pj-rehearse now and that there is a PR to get that working. I think it's this
one, right? https://github.com/openshift/ci-tools/pull/1430

I'm hoping that moving this job to step registry will possibly give us more logs in order to debug https://bugzilla.redhat.com/show_bug.cgi?id=1893348

**Comment by jluhrsen** [comment]
> @alvaroaleman , I think it's ready now, but it's failing pj-rehearse which I think is expected at this point. You noted in another PR I have that UNRESOLVED_CONFIG doesn't work for pj-rehearse now and that there is a PR to get that working. I think it's this
> one, right? [openshift/ci-tools#1430](https://github.com/openshift/ci-tools/pull/1430)
> 
> I'm hoping that moving this job to step registry will possibly give us more logs in order to debug https://bugzilla.redhat.com/show_bug.cgi?id=1893348

It looks like pj-rehearse is working for UNRESOLVED_CONFIG now, but I'm running in to something I don't understand. The
first rehearsal job failed because of some json config issue and I think it might be due to not putting quotes around the
version_bounds values. I updated the PR to use quotes, but the build_log is still showing it without quotes. The git checkout
file seems to be correct, so I can't explain it yet, but it seems like the rehearsal job is using an older version of this PR and not
the latest.

first job that ran when there were no quotes around version_bounds values:
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/13957/rehearse-13957-release-openshift-origin-installer-e2e-aws-upgrade-rollback-4.6-to-4.7/1333892772335194112

most recent job that looks the same to me:
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/13957/rehearse-13957-release-openshift-origin-installer-e2e-aws-upgrade-rollback-4.6-to-4.7/1333905879262564352

here is the current lines in the PR that have the quotes:
https://github.com/openshift/release/pull/13957/commits/4b5dcfb99c9c4e14370f587eddd0450e28db96f8#diff-f20da9fe69faf7e3bfa2be8f95367c6ef26e92e246cad6c78424807b153cf18fR5182

@alvaroaleman or @stevekuznetsov, any idea what I'm missing here?

File: ci-operator/step-registry/openshift/upgrade/rollback/aws/loki/openshift-upgrade-rollback-aws-loki-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE: "release:latest"
+    env:
+      TEST_COMMAND: "run-upgrade"
+      TEST_SUITE: "all"
+      DELETE_MC: "false"
+      TEST_OPTIONS: "abort-at=99"
+  documentation: |-
+    The Openshift E2E AWS workflow executes the upgrade end-to-end test suite on AWS with a default cluster
+    configuration and loki as a log collector, exercising an upgrade during the process and aborting it at
+    99% so that it will rollback to initial release.
```
Comment (vrutkovs) [inline ci-operator/step-registry/openshift/upgrade/rollback/aws/loki/openshift-upgrade-rollback-aws-loki-workflow.yaml:25]: `abort-at` here means "after 99 mins of waiting start an upgrade back to original release". Please update the docs accordingly.

**Review by vrutkovs — COMMENTED** [review]
I only skimmed through the PR, but how is it different from other `openshift-upgrade` workflows? Can we re-use the exising one and set TEST_OPTIONS instead?

File: ci-operator/step-registry/openshift/upgrade/rollback/aws/loki/openshift-upgrade-rollback-aws-loki-workflow.yaml (hunk 1 above)
Comment (jluhrsen) [reply ci-operator/step-registry/openshift/upgrade/rollback/aws/loki/openshift-upgrade-rollback-aws-loki-workflow.yaml:25]: I wasn't exactly sure what 99 quantified, but I did find this:
https://github.com/openshift/openshift-tests/blob/e0d37add24c5dfc34497ad334abc6f02be96312a/cmd/extended-platform-tests/openshift-tests.go#L158

is it 99m or 99%. still learning here.

**Comment by jluhrsen** [comment]
> I only skimmed through the PR, but how is it different from other `openshift-upgrade` workflows? Can we re-use the exising one and set TEST_OPTIONS instead?

Thanks for the review. Yes, I would totally rather re-use the normal upgrade workflows. I didn't think to find a way to put TEST_OPTIONS in the job definition and it would be overwritten in the upgrade workflow. If that works, it would be much better.
let me see if I can try that.

**Comment by jluhrsen** [comment]
> > I only skimmed through the PR, but how is it different from other `openshift-upgrade` workflows? Can we re-use the exising one and set TEST_OPTIONS instead?
> 
> Thanks for the review. Yes, I would totally rather re-use the normal upgrade workflows. I didn't think to find a way to put TEST_OPTIONS in the job definition and it would be overwritten in the upgrade workflow. If that works, it would be much better.
> let me see if I can try that.

@vrutkovs, how does it look now? I'm having trouble with the pj-rehearse job though. It seems like it's not actually using the changes from this PR to run that job.

**Comment by vrutkovs** [comment]
Looks good! IIRC `env` is being replaced, not merged, so check if copying all envs there works

**Review by AlexNPavel — APPROVED** [review]

**Pull request merged**